node/node.go

```diff
@@ -267,35 +268,48 @@ func createAndStartIndexerService(
 	dbProvider DBProvider,
 	eventBus *types.EventBus,
 	logger log.Logger,
-) (*indexer.Service, indexer.TxIndexer, indexer.BlockIndexer, error) {
-
-	var (
-		txIndexer    indexer.TxIndexer
-		blockIndexer indexer.BlockIndexer
-	)
+) (*indexer.Service, []indexer.EventSink, error) {
+
+	eventSinks := []indexer.EventSink{}
+	var sqlDB *sql.DB
+loop:
+	for _, db := range config.TxIndex.Indexer {
+		switch strings.ToLower(db) {
+		case string(indexer.NULL):
+			// when we see null in the config, the eventsinks will be reset with the nullEventSink.
+			eventSinks = append([]indexer.EventSink{}, nullSink.NewNullEventSink())
+			break loop
+		case string(indexer.KV):
+			store, err := dbProvider(&DBContext{"tx_index", config})
+			if err != nil {
+				return nil, nil, err
+			}
+			eventSinks = append(eventSinks, kvSink.NewKVEventSink(store))
+		case string(indexer.PSQL):
+			conn := config.TxIndex.PsqlConn
+			if conn == "" {
+				return nil, nil, errors.New("the psql connection settings cannot be empty")
+			}
 
-	switch config.TxIndex.Indexer {
-	case "kv":
-		store, err := dbProvider(&DBContext{"tx_index", config})
-		if err != nil {
-			return nil, nil, nil, err
+			es, db, err := psqlsink.NewPSQLEventSink(conn)
+			if err != nil {
+				return nil, nil, err
+			}
+			eventSinks = append(eventSinks, es)
+			sqlDB = db
+		default:
+			return nil, nil, errors.New("unsupported event sink type")
 		}
-
-		txIndexer = kv.NewTxIndex(store)
-		blockIndexer = blockidxkv.New(dbm.NewPrefixDB(store, []byte("block_events")))
-	default:
-		txIndexer = &null.TxIndex{}
-		blockIndexer = &blockidxnull.BlockerIndexer{}
 	}
 
-	indexerService := indexer.NewIndexerService(txIndexer, blockIndexer, eventBus)
+	indexerService := indexer.NewIndexerService(eventSinks, eventBus, sqlDB)
```

Comment:
We shouldn't need to directly pass a database object to the indexer service -- this seems like a codesmell.